src/transaction.js

```diff
@@ -17,13 +18,29 @@ import { Keypair } from './keypair';
  * submitting to the network or forwarding on to additional signers.
  * @constructor
  * @param {string|xdr.TransactionEnvelope} envelope - The transaction envelope object or base64 encoded string.
+ * @param {string} [networkPassphrase] passphrase of the target stellar network (e.g. "Public Global Stellar Network ; September 2015").
  */
 export class Transaction {
-  constructor(envelope) {
+  constructor(envelope, networkPassphrase) {
     if (typeof envelope === 'string') {
       const buffer = Buffer.from(envelope, 'base64');
       envelope = xdr.TransactionEnvelope.fromXDR(buffer);
     }
+
+    // Deprecation warning. TODO: remove optionality with next major release.
+    if (typeof networkPassphrase !== 'string') {
+      console.warn(
+        'Global `Network.current()` is deprecated. Please pass explicit argument instead, e.g. `new Transaction(envelope, Networks.PUBLIC)`.'
+      );
+      if (Network.current() === null) {
+        throw new Error(
+          'No network selected. Use `Network.use`, `Network.usePublicNetwork` or `Network.useTestNetwork` helper methods to select network.'
+        );
+      }
+      networkPassphrase = Network.current().networkPassphrase();
+    }
+    this.networkPassphrase = networkPassphrase;
```

Comment:
@Akuukis after thinking a little bit more about this, I'd prefer to avoid the breaking change. For now, I'd suggest adding a getter and use that to get `this._networkPassphrase` or read from Network. 

The code above will probably become something like

```
if (typeof networkPassphrase !== 'string') {
  console.warn(
    'Global `Network.current()` is deprecated. Please pass explicit argument instead, e.g. `new Transaction(envelope, Networks.PUBLIC)`.'
  );
}
this._networkPassphrase = networkPassphrase;
```

And the getter would be something like 

```
get networkPassphrase() {
  if (this._networkPassphrase) {
    return this._networkPassphrase;
  }

  console.warn(
    'Global `Network.current()` is deprecated. Please pass explicit argument instead, e.g. `new Transaction(envelope, Networks.PUBLIC)`.'
  );

  if (Network.current() === null) {
    throw new Error(
      'No network selected. Use `Network.use`, `Network.usePublicNetwork` or `Network.useTestNetwork` helper methods to select network.'
    );
  }

  return Network.current().networkPassphrase();
}
```